### ISSUE TITLE
[BUG]Service fails to start after reboot

### ISSUE THREAD

**Issue by weizhuangzhi** [bunkerity/bunkerweb#615]
**Description**
After installation under fedora38, it runs normally, but if the server is restarted, bunkerweb.service will fail to start.

**Logs**
```
[root@fedora38 ~]# systemctl status bunkerweb
× bunkerweb.service - BunkerWeb service
     Loaded: loaded (/usr/lib/systemd/system/bunkerweb.service; enabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: failed (Result: exit-code) since Tue 2023-09-05 16:22:56 CST; 1min 15s ago
   Duration: 5.113s
       Docs: https://docs.bunkerweb.io
    Process: 818 ExecStart=/usr/share/bunkerweb/scripts/start.sh start (code=exited, status=1/FAILURE)
   Main PID: 818 (code=exited, status=1/FAILURE)
        CPU: 1.012s

9月 05 16:22:51 fedora38 start.sh[818]: [2023-09-05 16:22:51] - SYSTEMCTL - ℹ️  - Scheduler already stopped
9月 05 16:22:51 fedora38 start.sh[818]: [2023-09-05 16:22:51] - SYSTEMCTL - ℹ️  - nginx is stopped
9月 05 16:22:52 fedora38 sudo[832]:     root : PWD=/ ; USER=nginx ; GROUP=nginx ; COMMAND=/bin/bash -c 'echo -ne \'IS_LOADING=yes\\nUSE_BUNKERNET=no\\nHTTP_>
9月 05 16:22:53 fedora38 sudo[959]:     root : PWD=/ ; USER=nginx ; GROUP=nginx ; COMMAND=/bin/bash -c 'PYTHONPATH=/usr/share/bunkerweb/deps/python/ /usr/sh>
9月 05 16:22:54 fedora38 start.sh[818]: [2023-09-05 16:22:54] - SYSTEMCTL - ℹ️ - Starting temp nginx ...
9月 05 16:22:56 fedora38 start.sh[994]: nginx: [emerg] open() "/var/run/bunkerweb/nginx.pid" failed (2: No such file or directory)
9月 05 16:22:56 fedora38 start.sh[818]: [2023-09-05 16:22:56] - SYSTEMCTL - ❌ - Error while executing temp nginx
9月 05 16:22:56 fedora38 systemd[1]: bunkerweb.service: Main process exited, code=exited, status=1/FAILURE
9月 05 16:22:56 fedora38 systemd[1]: bunkerweb.service: Failed with result 'exit-code'.
9月 05 16:22:56 fedora38 systemd[1]: bunkerweb.service: Consumed 1.012s CPU time.
```

**Temporary solution**
```
[root@fedora38 ~]# mkdir /run/bunkerweb/
chown -R nginx:nginx /run/bunkerweb/
[root@fedora38 ~]# systemctl start bunkerweb
[root@fedora38 ~]# systemctl status bunkerweb
● bunkerweb.service - BunkerWeb service
     Loaded: loaded (/usr/lib/systemd/system/bunkerweb.service; enabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: active (running) since Tue 2023-09-05 16:29:45 CST; 6s ago
       Docs: https://docs.bunkerweb.io
   Main PID: 1060 (start.sh)
      Tasks: 8 (limit: 4632)
     Memory: 52.6M
        CPU: 683ms
     CGroup: /system.slice/bunkerweb.service
             ├─1060 /bin/bash /usr/share/bunkerweb/scripts/start.sh start
             ├─1125 "nginx: master process nginx -e /var/log/bunkerweb/error.log"
             ├─1126 "nginx: worker process"
             ├─1127 "nginx: worker process"
             ├─1128 "nginx: worker process"
             ├─1130 "nginx: worker process"
             ├─1132 "nginx: worker process"
             └─1133 "nginx: worker process"

9月 05 16:29:45 fedora38 systemd[1]: Started bunkerweb.service - BunkerWeb service.
9月 05 16:29:45 fedora38 start.sh[1060]: [2023-09-05 16:29:45] - SYSTEMCTL - ℹ️ - Starting BunkerWeb service ...
9月 05 16:29:45 fedora38 start.sh[1060]: [2023-09-05 16:29:45] - SYSTEMCTL - ℹ️  - Scheduler already stopped
9月 05 16:29:45 fedora38 start.sh[1060]: [2023-09-05 16:29:45] - SYSTEMCTL - ℹ️  - nginx is stopped
9月 05 16:29:45 fedora38 sudo[1073]:     root : PWD=/ ; USER=nginx ; GROUP=nginx ; COMMAND=/bin/bash -c 'echo -ne \'IS_LOADING=yes\\nUSE_BUNKERNET=no\\nHTTP>
9月 05 16:29:45 fedora38 sudo[1086]:     root : PWD=/ ; USER=nginx ; GROUP=nginx ; COMMAND=/bin/bash -c 'PYTHONPATH=/usr/share/bunkerweb/deps/python/ /usr/s>
9月 05 16:29:46 fedora38 start.sh[1060]: [2023-09-05 16:29:46] - SYSTEMCTL - ℹ️ - Starting temp nginx ...
```

After each restart, you need to create the directory and set the owner service to start and run normally. Is this problem because that's how bunkerweb is designed or am I doing something wrong?

**Comment by TheophileDiot** [comment]
@weizhuangzhi Could you try reproducing the error and then show me the content of this file `/var/log/bunkerweb/error.log` ?

**Comment by weizhuangzhi** [comment]
```
2023/09/06 08:55:43 [notice] 1289#1289: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:55:43 [notice] 1289#1289: exiting
2023/09/06 08:55:43 [notice] 1286#1286: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:55:43 [notice] 1289#1289: signal 1 (SIGHUP) received from 1, ignoring
2023/09/06 08:55:43 [notice] 1288#1288: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:55:43 [notice] 1288#1288: signal 1 (SIGHUP) received from 1, ignoring
2023/09/06 08:55:43 [notice] 1288#1288: exiting
2023/09/06 08:55:43 [notice] 1286#1286: signal 1 (SIGHUP) received from 1, reconfiguring
2023/09/06 08:55:43 [notice] 1287#1287: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:55:43 [notice] 1287#1287: signal 1 (SIGHUP) received from 1, ignoring
2023/09/06 08:55:43 [notice] 1287#1287: exiting
2023/09/06 08:55:43 [notice] 1292#1292: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:55:43 [notice] 1290#1290: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:55:43 [notice] 1292#1292: signal 1 (SIGHUP) received from 1, ignoring
2023/09/06 08:55:43 [notice] 1290#1290: signal 1 (SIGHUP) received from 1, ignoring
2023/09/06 08:55:43 [notice] 1290#1290: exiting
2023/09/06 08:55:43 [notice] 1292#1292: exiting
2023/09/06 08:55:43 [notice] 1291#1291: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:55:43 [notice] 1291#1291: signal 1 (SIGHUP) received from 1, ignoring
2023/09/06 08:55:43 [notice] 1291#1291: exiting
2023/09/06 08:55:43 [notice] 1286#1286: signal 14 (SIGALRM) received
2023/09/06 08:55:43 [notice] 1288#1288: exit
2023/09/06 08:55:43 [notice] 1291#1291: exit
2023/09/06 08:55:43 [notice] 1289#1289: exit
2023/09/06 08:55:43 [notice] 1286#1286: signal 17 (SIGCHLD) received from 1288
2023/09/06 08:55:43 [notice] 1286#1286: worker process 1288 exited with code 0
2023/09/06 08:55:43 [notice] 1286#1286: signal 29 (SIGIO) received
2023/09/06 08:55:43 [notice] 1292#1292: exit
2023/09/06 08:55:43 [notice] 1286#1286: signal 17 (SIGCHLD) received from 1291
2023/09/06 08:55:43 [notice] 1286#1286: worker process 1291 exited with code 0
2023/09/06 08:55:43 [notice] 1286#1286: signal 29 (SIGIO) received
2023/09/06 08:55:43 [notice] 1286#1286: signal 17 (SIGCHLD) received from 1289
2023/09/06 08:55:43 [notice] 1286#1286: worker process 1289 exited with code 0
2023/09/06 08:55:43 [notice] 1286#1286: signal 29 (SIGIO) received
2023/09/06 08:55:43 [notice] 1286#1286: signal 17 (SIGCHLD) received from 1292
2023/09/06 08:55:43 [notice] 1286#1286: worker process 1292 exited with code 0
2023/09/06 08:55:43 [notice] 1286#1286: signal 29 (SIGIO) received
2023/09/06 08:55:43 [notice] 1290#1290: exit
2023/09/06 08:55:43 [notice] 1287#1287: exit
2023/09/06 08:55:43 [notice] 1286#1286: signal 17 (SIGCHLD) received from 1290
2023/09/06 08:55:43 [notice] 1286#1286: worker process 1290 exited with code 0
2023/09/06 08:55:43 [notice] 1286#1286: signal 29 (SIGIO) received
2023/09/06 08:55:44 [notice] 1286#1286: signal 17 (SIGCHLD) received from 1287
2023/09/06 08:55:44 [notice] 1286#1286: worker process 1287 exited with code 0
2023/09/06 08:55:44 [notice] 1286#1286: exit
2023/09/06 08:56:13 [notice] 993#993: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/0/0)
2023/09/06 08:56:13 [notice] 993#993: [INIT] init phase started
2023/09/06 08:56:13 [notice] 993#993: [INIT] deleting old keys from datastore ...
2023/09/06 08:56:13 [notice] 993#993: [INIT] deleted old keys from datastore
2023/09/06 08:56:13 [notice] 993#993: [INIT] saving plugins into datastore ...
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin antibot v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin authbasic v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin badbehavior v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin blacklist v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin brotli v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin bunkernet v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin clientcache v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin cors v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin country v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin customcert v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin db v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin dnsbl v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin errors v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin greylist v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin gzip v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin headers v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin inject v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin jobs v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin letsencrypt v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin limit v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin misc v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin modsecurity v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin php v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin realip v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin redirect v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin redis v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin reverseproxy v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin reversescan v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin selfsigned v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin sessions v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin ui v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] loaded plugin whitelist v1.0
2023/09/06 08:56:13 [notice] 993#993: [INIT] saving variables into datastore ...
2023/09/06 08:56:13 [notice] 993#993: [INIT] saved variables into datastore
2023/09/06 08:56:13 [notice] 993#993: [INIT] saving plugins order into datastore ...
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase log : ["bunkernet","badbehavior"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase set : ["sessions","whitelist"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase access : ["whitelist","letsencrypt","blacklist","greylist","country","dnsbl","bunkernet","reversescan","limit","misc","cors","antibot"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase init : ["sessions","whitelist","blacklist","greylist","bunkernet","limit","headers"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase header : ["cors","antibot","headers"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase init_worker : ["redis","bunkernet","dnsbl"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase preread : ["whitelist","blacklist","greylist","country","dnsbl","reversescan"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase log_stream : ["badbehavior","bunkernet"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] plugins order for phase log_default : ["bunkernet","badbehavior"]
2023/09/06 08:56:13 [notice] 993#993: [INIT] saved plugins order into datastore
2023/09/06 08:56:13 [notice] 993#993: [INIT] calling init() methods of plugins ...
2023/09/06 08:56:13 [notice] 993#993: [INIT] sessions:init() call successful : init not needed
2023/09/06 08:56:13 [notice] 993#993: [INIT] whitelist:init() call successful : init not needed
2023/09/06 08:56:13 [notice] 993#993: [INIT] blacklist:init() call successful : init not needed
2023/09/06 08:56:13 [notice] 993#993: [INIT] greylist:init() call successful : init not needed
2023/09/06 08:56:13 [notice] 993#993: [INIT] bunkernet:init() call successful : no service uses BunkerNet, skipping init
2023/09/06 08:56:13 [notice] 993#993: [INIT] limit:init() call successful : no service uses limit for requests, skipping init
2023/09/06 08:56:13 [notice] 993#993: [INIT] headers:init() call successful : successfully loaded 1 custom headers
2023/09/06 08:56:13 [notice] 993#993: [INIT] called init() methods of plugins
2023/09/06 08:56:13 [notice] 993#993: [INIT] init phase ended
2023/09/06 08:56:13 [notice] 993#993: using the "epoll" event method
2023/09/06 08:56:13 [notice] 993#993: nginx/1.24.0
2023/09/06 08:56:13 [notice] 993#993: OS: Linux 6.4.13-200.fc38.x86_64
2023/09/06 08:56:13 [notice] 993#993: getrlimit(RLIMIT_NOFILE): 1024:524288
2023/09/06 08:56:13 [emerg] 996#993: open() "/var/run/bunkerweb/nginx.pid" failed (2: No such file or directory)
2023/09/06 08:56:13 [notice] 996#996: start worker processes
2023/09/06 08:56:13 [notice] 996#996: start worker process 997
2023/09/06 08:56:13 [notice] 996#996: start worker process 998
2023/09/06 08:56:13 [notice] 996#996: start worker process 999
2023/09/06 08:56:13 [notice] 996#996: start worker process 1000
2023/09/06 08:56:13 [notice] 996#996: start worker process 1001
2023/09/06 08:56:13 [notice] 996#996: start worker process 1002
2023/09/06 08:56:15 [notice] 996#996: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:56:15 [notice] 997#997: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:56:15 [notice] 998#998: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:56:15 [notice] 999#999: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:56:15 [notice] 997#997: exiting
2023/09/06 08:56:15 [notice] 998#998: exiting
2023/09/06 08:56:15 [notice] 999#999: exiting
2023/09/06 08:56:15 [notice] 1002#1002: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:56:15 [notice] 1002#1002: exiting
2023/09/06 08:56:15 [notice] 1001#1001: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:56:15 [notice] 1001#1001: exiting
2023/09/06 08:56:15 [notice] 1000#1000: signal 15 (SIGTERM) received from 1, exiting
2023/09/06 08:56:15 [notice] 1000#1000: exiting
2023/09/06 08:56:15 [notice] 1002#1002: exit
2023/09/06 08:56:15 [notice] 1000#1000: exit
2023/09/06 08:56:15 [notice] 998#998: exit
2023/09/06 08:56:15 [notice] 997#997: exit
2023/09/06 08:56:15 [notice] 999#999: exit
2023/09/06 08:56:15 [notice] 1001#1001: exit
2023/09/06 08:56:15 [notice] 996#996: signal 14 (SIGALRM) received
2023/09/06 08:56:15 [notice] 996#996: signal 17 (SIGCHLD) received from 1002
2023/09/06 08:56:15 [notice] 996#996: worker process 1002 exited with code 0
2023/09/06 08:56:15 [notice] 996#996: signal 29 (SIGIO) received
2023/09/06 08:56:15 [notice] 996#996: signal 17 (SIGCHLD) received from 997
2023/09/06 08:56:15 [notice] 996#996: worker process 997 exited with code 0
2023/09/06 08:56:15 [notice] 996#996: signal 29 (SIGIO) received
2023/09/06 08:56:15 [notice] 996#996: signal 17 (SIGCHLD) received from 1000
2023/09/06 08:56:15 [notice] 996#996: worker process 1000 exited with code 0
2023/09/06 08:56:15 [notice] 996#996: signal 29 (SIGIO) received
2023/09/06 08:56:15 [notice] 996#996: signal 17 (SIGCHLD) received from 999
2023/09/06 08:56:15 [notice] 996#996: worker process 999 exited with code 0
2023/09/06 08:56:15 [notice] 996#996: worker process 1001 exited with code 0
2023/09/06 08:56:15 [notice] 996#996: signal 29 (SIGIO) received
2023/09/06 08:56:15 [notice] 996#996: signal 17 (SIGCHLD) received from 998
2023/09/06 08:56:15 [notice] 996#996: worker process 998 exited with code 0
2023/09/06 08:56:15 [alert] 996#996: unlink() "/var/run/bunkerweb/nginx.pid" failed (2: No such file or directory)
2023/09/06 08:56:15 [notice] 996#996: exit
```

@TheophileDiot The above logs are the contents of/var/log/bunkerweb/error.log after reboot server

**Comment by TheophileDiot** [comment]
Hi @weizhuangzhi, I tried to reproduce your bug without success, could you provide us with more information like what OS you are using (the exact version of fedora for example), the version of BunkerWeb and also your configuration ? If that is alright with you, thank you !

**Comment by weizhuangzhi** [comment]
Fedora：[Fedora-Server-38-1.6](https://download.fedoraproject.org/pub/fedora/linux/releases/38/Server/x86_64/iso/Fedora-Server-dvd-x86_64-38-1.6.iso)

BunkerWeb：1.5.1

[Integrations](https://docs.bunkerweb.io/1.5.1/integrations/)  follow the official documentation

variables.env
```
DNS_RESOLVERS=8.8.8.8 8.8.4.4
HTTP_PORT=80
HTTPS_PORT=443
API_LISTEN_IP=127.0.0.1
SERVER_NAME=192.168.144.6
MULTISITE=yes

192.168.144.6_USE_UI=yes
192.168.144.6_USE_REVERSE_PROXY=yes
192.168.144.6_REVERSE_PROXY_URL=/manage
192.168.144.6_REVERSE_PROXY_HOST=http://127.0.0.1:7000
192.168.144.6_INTERCEPTED_ERROR_CODES=400 404 405 413 429 500 501 502 503 504
```

Do you need additional information?

**Comment by TheophileDiot** [comment]
Thank you for these information, I'll run some tests and let you know !

**Comment by TheophileDiot** [comment]
I tried using the latest Fedora server version with BunkerWeb 1.5.1 (I also followed our doc exactly) and I got no Issues, Can you try updating your server's version maybe and reinstalling BunkerWeb after a `purge` ?

**Comment by weizhuangzhi** [comment]
I'll try again. 
After the installation is complete, the service can be started directly.
Have you reboot the operating system? I can't start the service after reboot the operating system.

**Comment by TheophileDiot** [comment]
I didn't try this, let me check

**Comment by TheophileDiot** [comment]
Okay, I got the same issue as you, I'll investigate further more and let you know.

**Comment by TheophileDiot** [comment]
It turns out that fedora deletes the `/var/run/bunkerweb` when rebooting, we'll add a check that fixes this, thank you again for this issue !

**Comment by fl0ppy-d1sk** [comment]
Hello @weizhuangzhi,

It should be fixed in the last [1.5.2 version](https://github.com/bunkerity/bunkerweb/releases/tag/v1.5.2). Feel free to open a new issue if needed.
